recipes/paho-mqtt-cpp/all/conanfile.py

```diff
@@ -83,6 +83,9 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
 
     def package_info(self):
-        self.cpp_info.libs = tools.collect_libs(self)
         self.cpp_info.names["cmake_find_package"] = "PahoMqttCpp"
         self.cpp_info.names["cmake_find_package_multi"] = "PahoMqttCpp"
+        cmake_target = "paho-mqttpp3" if self.options.shared else "paho-mqttpp3-static"
+        self.cpp_info.components["_paho-mqtt-cpp"].names["cmake_find_package"] = cmake_target
+        self.cpp_info.components["_paho-mqtt-cpp"].names["cmake_find_package_multi"] = cmake_target
+        self.cpp_info.components["_paho-mqtt-cpp"].libs = tools.collect_libs(self)
```

Comment:
Does this mean that i need to update my cmake scripts depending on if I’m building (or using) a shared to static library?

Reply:
Yes, if you want transparent cmake integration.
It's worth noting, that `cmake_find_package` and `cmake_find_package_multi` still create `PahoMqttCpp::PahoMqttCpp` (unofficial) target.

Reply:
poh, its a pain that people (upstream) sill doing this... they could also start creating targets depending on the compiler, standard, with and without ssl....

Reply:
Loot at `paho-mqtt-c` and cry :sweat_smile: https://github.com/conan-io/conan-center-index/pull/2729